### PR TITLE
Tweak: Tools - Change font awesome migration process [ED-6935]

### PR DESCRIPTION
**Pull request by Nevoss**

### CODE REVIEW

File: includes/managers/icons.php [hunk 1]
```diff
@@ -516,6 +517,29 @@ public function register_admin_tools_settings( Tools $settings ) {
 		] );
 	}
 
+	/**
+	 * Get redirect URL when uprating font awesome.
```
Comment (StyleShit) [inline includes/managers/icons.php:521]:
```suggestion
	 * Get redirect URL when upgrading font awesome.
```

File: includes/managers/icons.php [hunk 2]
```diff
@@ -502,12 +502,13 @@ public function register_admin_tools_settings( Tools $settings ) {
 			},
 			'fields' => [
 				[
-					'label'      => esc_html__( 'Font Awesome Upgrade', 'elementor' ),
+					'label' => esc_html__( 'Font Awesome Upgrade', 'elementor' ),
 					'field_args' => [
 						'type' => 'raw_html',
-						'html' => sprintf( '<span data-action="%s" data-_nonce="%s" class="button" id="elementor_upgrade_fa_button">%s</span>',
+						'html' => sprintf( '<span data-action="%s" data-_nonce="%s" data-redirect-url="%s" class="button" id="elementor_upgrade_fa_button">%s</span>',
```
Comment (StyleShit) [inline includes/managers/icons.php:508]: it's not a "redirect URL" anymore. Maybe change the name to something else that suggests it's a document ID?

File: includes/managers/icons.php [hunk 3]
```diff
@@ -516,6 +517,29 @@ public function register_admin_tools_settings( Tools $settings ) {
...
+	public function get_upgrade_redirect_url() {
+		// PHPCS - No need to verify nonce, the 'redirect_to' could be only post id, and if it is not the
+		// URL will be empty.
+		$document_id = ! empty( $_GET['redirect_to'] ) ? $_GET['redirect_to'] : null; // phpcs:ignore
+
+		if ( ! is_numeric( $document_id ) ) {
+			return '';
+		}
+
+		$document = Plugin::$instance->documents->get( intval( $document_id ) );
```
Comment (StyleShit) [inline includes/managers/icons.php:534]:
```suggestion
		$document_id = ! empty( $_GET['redirect_to'] ) ? (int) $_GET['redirect_to'] : null; // phpcs:ignore

		if ( ! $document_id ) {
			return '';
		}

		$document = Plugin::$instance->documents->get( $document_id );
```

**Review by StyleShit — CHANGES_REQUESTED** [review]

File: includes/managers/icons.php (hunk 1 above)
Comment (Nevoss) [reply includes/managers/icons.php:521]: Done

File: includes/managers/icons.php (hunk 2 above)
Comment (Nevoss) [reply includes/managers/icons.php:508]: Done

File: includes/managers/icons.php (hunk 3 above)
Comment (Nevoss) [reply includes/managers/icons.php:534]: Done

**Review by StyleShit — APPROVED** [review]

File: includes/managers/icons.php [hunk 4]
```diff
@@ -516,6 +517,29 @@ public function register_admin_tools_settings( Tools $settings ) {
...
 
+	/**
+	 * Get redirect URL when upgrading font awesome.
+	 *
+	 * @return string
+	 */
+	public function get_upgrade_redirect_url() {
+		// PHPCS - No need to verify nonce, the 'redirect_to' could be only post id, and if it is not the
+		// URL will be empty.
+		$document_id = ! empty( $_GET['redirect_to_document'] ) ? $_GET['redirect_to_document'] : null; // phpcs:ignore
```
Comment (KingYes) [inline includes/managers/icons.php:528]: Please declare the specific role you want to ignore.

File: includes/managers/icons.php [hunk 5]
```diff
@@ -516,6 +517,29 @@ public function register_admin_tools_settings( Tools $settings ) {
...
+
+		if ( ! is_numeric( $document_id ) ) {
```
Comment (KingYes) [inline includes/managers/icons.php:530]: Why you don'y using `absint()` for the validation?

**Review by KingYes — CHANGES_REQUESTED** [review]

File: includes/managers/icons.php (hunk 4 above)
Comment (Nevoss) [reply includes/managers/icons.php]: Done

File: includes/managers/icons.php (hunk 5 above)
Comment (Nevoss) [reply includes/managers/icons.php]: Done

File: includes/managers/icons.php [hunk 6]
```diff
@@ -516,6 +517,29 @@ public function register_admin_tools_settings( Tools $settings ) {
 		] );
 	}
 
+	/**
+	 * Get redirect URL when upgrading font awesome.
+	 *
+	 * @return string
+	 */
+	public function get_upgrade_redirect_url() {
+		// PHPCS - No need to verify nonce, the 'redirect_to' could be only post id, and if it is not the
```
Comment (matipojo) [inline includes/managers/icons.php:526]: Why not adding a nonce?

**Review by matipojo — CHANGES_REQUESTED** [review]

File: includes/managers/icons.php (hunk 6 above)
Comment (Nevoss) [reply includes/managers/icons.php:526]: Seems like there is no need for the nonce, only the ID is transferred through the GET params.

Comment (matipojo) [reply includes/managers/icons.php:526]: It may be changed over time.

Comment (Nevoss) [reply includes/managers/icons.php]: Done

**Review by matipojo — APPROVED** [review]

**Pull request merged**